rosbag2_compression/test/rosbag2_compression/test_compression_factory.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <string>
+#include <algorithm>
+
+#include "rosbag2_compression/compression_factory.hpp"
+
+namespace
+{
+
+std::string to_lower(const std::string & text)
+{
+  std::string lowercase_text = text;
+  std::transform(lowercase_text.begin(), lowercase_text.end(), lowercase_text.begin(), ::tolower);
+  return lowercase_text;
+}
+}  // namespace
+
+class CompressionFactoryTest : public ::testing::Test
+{
+public:
+  rosbag2_compression::CompressionFactory factory{};
+};
+
+TEST_F(CompressionFactoryTest, creates_zstd_compressor) {
+  const auto compression_format = "zstd";
+  auto zstd_compressor = factory.create_compressor(compression_format);
+  ASSERT_EQ(compression_format, zstd_compressor->get_compression_identifier());
+}
+
+TEST_F(CompressionFactoryTest, creates_zstd_compressor_caseinsensitive)
+{
+  const auto compression_format = "ZsTd";
+  const auto lowercase_compression_format = to_lower(compression_format);
```

Comment:
Why reimplement ```case_insensitive_compare```?

Reply:
Not sure I follow this one. I need to verify that the compressor created is the correct one.
Are you suggesting that I instead use `case_insensitive_compare` to compare the compression format?

Reply:
This might be resolved by making `get_compression_identifier()` static as well.